src/router/bundle/bundleHandler.ts

```diff
@@ -146,6 +147,32 @@ export default class BundleHandler implements BundleHandlerInterface {
             }
         }
 
+        const authAndFilterReadPromises = requests.map((request, index) => {
+            if (['read', 'vread', 'history-type', 'history-instance', 'search-type'].includes(request.operation)) {
+                return this.authService.authorizeAndFilterReadResponse({
+                    operation: request.operation,
+                    userIdentity,
+                    readResponse: bundleServiceResponse.batchReadWriteResponses[index].resource,
+                });
+            }
+            return Promise.resolve();
+        });
+
+        const readResponses = await Promise.allSettled(authAndFilterReadPromises);
+
+        requests.forEach((request, index) => {
+            const responseEntry = bundleServiceResponse.batchReadWriteResponses[index];
+            if (['read', 'vread', 'history-type', 'history-instance', 'search-type'].includes(request.operation)) {

```

Comment:
Do we need to include system reads?

Reply:
Are system reads, `search-system` and `history-system`? 
https://github.com/awslabs/fhir-works-on-aws-interface/blob/mainline/src/constants.ts#L70

In our implementation of Bundles we only support CRUD requests inside the Bundle, but I'll add `system-search` and `history-system` to this array for completeness.